realm/realm-library/src/androidTest/java/io/realm/OrderedCollectionChangeSetTests.java

```diff
@@ -130,11 +124,30 @@ private void deleteObjects(Realm realm, int... indices) {
     // Creates Dogs objects with columnLong set to the value elements in indices array.
     private void createObjects(Realm realm, int... indices) {
         for (int index : indices) {
-            realm.createObject(Dog.class).setAge(index);
+            Dog dog = realm.createObject(Dog.class);
+            dog.setAge(index);
+            if (type == ObservablesType.REALM_LIST) {
+                Owner owner = realm.where(Owner.class).findFirst();
+                assertNotNull(owner);
+                RealmList<Dog> dogs = owner.getDogs();
+                boolean added = false;
+                for (int i = 0; i < dogs.size(); i++) {
+                    if (dog.getAge() <= dogs.get(i).getAge()) {
```

Comment:
This could use a little description I think. There looks like there is some assumptions about the `index` and `dogs.size()` that I don't fully understand.

Reply:
added some comments. The assumption is the both `RealmResults` and `RealmList` are sorted by age. See `populateData()`.